src/org/ods/component/RolloutOpenShiftDeploymentStage.groovy

```diff
@@ -94,28 +90,51 @@ class RolloutOpenShiftDeploymentStage extends Stage {
                     )
                 }
             }
-            // If deployment did not exist prior to this stage,
-            // check again now as Tailor might have created it.
-            if (!dExists) {
-                dExists = deploymentExists()
-            }
         }
 
-        if (!dExists) {
-            script.error "${config.deploymentKind} '${config.resourceName}' does not exist. " +
-                'Verify that you have setup the OpenShift resource correctly ' +
-                'and/or set the "deploymentKind"/"resourceName" option of the pipeline stage appropriately.'
+        deploymentResources = openShift.getResourcesForComponent(
+            context.targetProject, deploymentKinds, config.selector
+        )
+        return rollout(deploymentResources, originalDeploymentVersions)
+    }
+
+    private Map<String,List<Map>> rollout(
+        Map<String, List<String>> deploymentResources,
+        Map<String, Map<String, Integer>> originalVersions) {
+        def rolloutData = [:]
+        deploymentResources.each { resourceKind, resourceNames ->
+            resourceNames.each { resourceName ->
+                def originalVersion = 0
+                if (originalVersions.containsKey(resourceKind)) {
+                    originalVersion = originalVersions[resourceKind][resourceName] ?: 0
+                }
+
+                def podData = rolloutDeployment(resourceKind, resourceName, originalVersion)
+
+                if (!rolloutData.containsKey(resourceKind)) {
+                    rolloutData[resourceKind] = [:]
+                }
+                rolloutData[resourceKind][resourceName] = podData
+                // TODO: Once the orchestration pipeline can deal with multiple replicas,
+                // update this to store multiple pod artifacts.
+                // TODO: Potential conflict if resourceName is duplicated between
```

Comment:
Should we add a warning message for this case? although it could be checked beforehand also.
Thinking more on this edge case, shall we just fail because it shouldn't happen?

Reply:
I thought about this but have not yet reached a conclusion how to deal with it / where to fail. Maybe this can also be solved by making an update to the orchestration pipeline (as in, where it expects artefacts). Therefore I'm going to leave this as is, and revisit later.